### DISCUSSION TITLE
Help with wiring for tasmota flash

### DISCUSSION THREAD

**Discussion by broyuken** [arendst/Tasmota#17152] [Q&A] [answered]
Hi all, about to embark on my first non Tuya convert flash. I could use some help identifying where I need to connect to. This is the novastella floodlight pictured here. 

https://templates.blakadder.com/novostella_UT88835.html

I followed the instructions from the above site and dug out the esp, but the pins are a little confusing to me. Here are pics of the left and right side of the vertical chip and the bottom. 3.3 and ground are pretty clear but rx and tx are where I’m getting stuck. I’m assuming tx is one of the tiny pins above the tx writing on the left hand side, but does it matter which pin? And rx I have no ideas on. Help please!

![89263D23-FFA3-41A0-9790-57362820B7E6](https://user-images.githubusercontent.com/1593391/202936899-b03c52ce-8603-4cf4-9e06-2e5cca799569.jpeg)

![4382E75C-9972-47FC-A970-1035FA2CBD63](https://user-images.githubusercontent.com/1593391/202936892-a03e0bc7-7c3d-4050-a7ce-c54bb67c841d.jpeg)
![FFC6ED8F-F33F-4FCB-B02B-ECE4F83660C6](https://user-images.githubusercontent.com/1593391/202936895-2d34a8d7-8398-4be6-8efc-bd6c5589d5f6.jpeg)

**Comment by barbudor** [comment]
The pins should be on the ESP module, or where the module connects to the main PCB. 
Have you checked the photos of the module on the tuya site: https://developer.tuya.com/en/docs/iot/wifilc4module?id=K9605t0d19jiy#title-3-Module%20picture
Does it match yours? 
Time to confirm there is an Espressif logo on the chip. 

The only complicated one is IO0 as it is on the back of the module

**Reply by broyuken** [reply]
Yes it does have the esp8266 on it. And it does look like that diagram. 

![9BD26A71-C739-4A7A-8DD4-1218502D52B0](https://user-images.githubusercontent.com/1593391/203084356-37483ff0-4b75-4caa-8e79-3460e24db13c.jpeg)

is the best way to do it to match up the pins underneath the chip? Where it meets the other board?

**Comment by broyuken** [comment, marked as answer]
If I'm reading the diagram correct it would be like this right? Also I've cleaned up those pins since I took the picture :)

![image](https://user-images.githubusercontent.com/1593391/203088833-a778f2d7-69aa-4509-bc1f-f031fde92c86.png)

**Reply by barbudor** [reply]
I would say the same as you 👍 
Don't forget you need to ground IO0 which is on the back of the ESP module:

![image](https://user-images.githubusercontent.com/2996042/203125331-d1290164-1e9b-44f9-8b95-798bf743c98c.png)

**Reply by broyuken** [reply]
Yep, it won't upload without that. And I believe I read that it's ok to stay grounded throughout the whole flashing process. And that would just involve holding or soldering a wire from IO0 to the GND pin if I'm not mistaken.

**Reply by barbudor** [reply]
Exactly
It's mandatory at the beginning but not a problem if it's stay grounded during the whole process

**Reply by broyuken** [reply]
thank you so much! As soon as my CH340 comes in I'll give it a go!

**Comment by broyuken** [comment]
Just an update, the flash worked fine, but the light itself is dead. It just keeps strobing. Something else must be wrong with it hardware wise.
